ecommerce/backend/API/views/order.py

```diff
@@ -0,0 +1,72 @@
+from rest_framework import exceptions
+from rest_framework.response import Response
+from rest_framework.decorators import api_view, permission_classes
+from rest_framework import status
+import json
+
+from ..utils import permissions, Role
+from ..models import Product,Vendor,ImageUrls,Category,Subcategory,Purchase
+from ..serializers.address_serializer import AddressResponseSerializer
+from ..serializers.product_serializer import ProductResponseSerializer
+from ..utils.order_status import OrderStatus
+
+
+@api_view(['GET','PUT'])
+@permission_classes([permissions.IsVendorUser]) 
+def vendor_orders(request):
+
+    if request.method == 'GET': 
+
+        response_purchases = []
+
+        vendor = Vendor.objects.filter(user=request.user).first() 
+
+        my_purchases = Purchase.objects.filter(vendor_id=vendor.id) 
+
+        for purchase in my_purchases:
+
+            total_price = purchase.amount*purchase.unit_price
+
+            serializer = AddressResponseSerializer(purchase.address)
+            adress = serializer.data
+
+            serializer = ProductResponseSerializer(purchase.product)
+            product = serializer.data
+
+            response = {'amount':purchase.amount,'unit_price':purchase.unit_price,'total_price':total_price,'status':purchase.status,
+            'purchase_date':purchase.purchase_date,'adress':adress, 'product':product, 'id':purchase.id}
+
+            response_purchases.append(response)
+
+        return Response({'orders':response_purchases})
```

Comment:
Can you format all of your responses as either one of
- `{{'successful': True, 'message':'Successfully retrieved'}, 'orders': []}  
- {{'successful': True, 'message':'Successfully retrieved'}, 'order': {}}  
- {{'successful': False, 'message':'Error occurred'}}  
`

Reply:
That is the backend team spirit ❤️